### ISSUE TITLE
Could not resolve cz-conventional-changelog-for-jira from czrc and husky 7

### ISSUE THREAD

**Issue by neginqantas** [digitalroute/cz-conventional-changelog-for-jira#35]
https://stackoverflow.com/questions/68448567/cz-conventional-changelog-for-jira-set-up-with-husky-v7-is-not-working

`    "@digitalroute/cz-conventional-changelog-for-jira": "^6.5.0",
    "husky": "^6.0.0",`

After running git commit I will get:

`Could not resolve cz-conventional-changelog-for-jira. Cannot find module 'cz-conventional-changelog-for-jira'
Require stack:
- /Users/neginbasiri/Projects/web-ui/node_modules/commitizen/dist/commitizen/adapter.js
- /Users/neginbasiri/Projects/web-ui/node_modules/commitizen/dist/commitizen.js
- /Users/neginbasiri/Projects/web-ui/node_modules/commitizen/dist/cli/git-cz.js
- /Users/neginbasiri/Projects/web-ui/node_modules/commitizen/bin/git-cz.js
- /Users/neginbasiri/Projects/web-ui/node_modules/commitizen/bin/git-cz`

If I change path in .czrc from "path": "cz-conventional-changelog-for-jira", to "path": "cz-conventional-changelog",
It works perfectly. So something is wrong with the "cz-conventional-changelog-for-jira"

**Comment by juliuscc** [comment]
You have to do `"path": "@digitalroute/cz-conventional-changelog-for-jira"`

**Comment by juliuscc** [comment]
Could you provide a minimal reproducible repo if the issue still persists?

**Comment by neginqantas** [comment]
Thanks @juliuscc I changed the path and the error message now is

`> exec < /dev/tty && npx git-cz --hook || true

Cannot find module 'inquirer/lib/prompts/input'
Require stack:
- /Users/neginbasiri/Projects/web-ui/node_modules/@digitalroute/cz-conventional-changelog-for-jira/LimitedInputPrompt.js
- /Users/neginbasiri/Projects/web-ui/node_modules/@digitalroute/cz-conventional-changelog-for-jira/engine.js
- /Users/neginbasiri/Projects/web-ui/node_modules/@digitalroute/cz-conventional-changelog-for-jira/index.js
- /Users/neginbasiri/Projects/web-ui/node_modules/commitizen/dist/commitizen/adapter.js
- /Users/neginbasiri/Projects/web-ui/node_modules/commitizen/dist/commitizen.js
- /Users/neginbasiri/Projects/web-ui/node_modules/commitizen/dist/cli/git-cz.js
- /Users/neginbasiri/Projects/web-ui/node_modules/commitizen/bin/git-cz.js
- /Users/neginbasiri/Projects/web-ui/node_modules/commitizen/bin/git-cz`

**Comment by juliuscc** [comment]
Could you please provide a GitHub repo with this issue?

**Comment by laytong** [comment]
So this has started happening for me.

Looks like this happens when you're using yarn's Plug N Play

Presumably some underlying dependency should be declaring inquirer as a dependency

**Comment by laytong** [comment]
```@digitalroute/cz-conventional-changelog-for-jira tried to access inquirer, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

Required package: inquirer (via "inquirer/lib/prompts/input")
Required by: @digitalroute/cz-conventional-changelog-for-jira@npm:7.0.0 (via /Users/lgilbraith/code/dt-fleet-micro-fe-albus/.yarn/cache/@digitalroute-cz-conventional-changelog-for-jira-npm-7.0.0-1d935757fc-470f1ef69f.zip/node_modules/@digitalroute/cz-conventional-changelog-for-jira/)

Require stack:
- /Users/lgilbraith/code/dt-fleet-micro-fe-albus/.yarn/cache/@digitalroute-cz-conventional-changelog-for-jira-npm-7.0.0-1d935757fc-470f1ef69f.zip/node_modules/@digitalroute/cz-conventional-changelog-for-jira/LimitedInputPrompt.js
- /Users/lgilbraith/code/dt-fleet-micro-fe-albus/.yarn/cache/@digitalroute-cz-conventional-changelog-for-jira-npm-7.0.0-1d935757fc-470f1ef69f.zip/node_modules/@digitalroute/cz-conventional-changelog-for-jira/engine.js
- /Users/lgilbraith/code/dt-fleet-micro-fe-albus/.yarn/cache/@digitalroute-cz-conventional-changelog-for-jira-npm-7.0.0-1d935757fc-470f1ef69f.zip/node_modules/@digitalroute/cz-conventional-changelog-for-jira/index.js
- /Users/lgilbraith/code/dt-fleet-micro-fe-albus/.yarn/cache/commitizen-npm-4.2.4-6b34777726-5b0ae7310e.zip/node_modules/commitizen/dist/commitizen/adapter.js
- /Users/lgilbraith/code/dt-fleet-micro-fe-albus/.yarn/cache/commitizen-npm-4.2.4-6b34777726-5b0ae7310e.zip/node_modules/commitizen/dist/commitizen.js
- /Users/lgilbraith/code/dt-fleet-micro-fe-albus/.yarn/cache/commitizen-npm-4.2.4-6b34777726-5b0ae7310e.zip/node_modules/commitizen/dist/cli/git-cz.js
- /Users/lgilbraith/code/dt-fleet-micro-fe-albus/.yarn/cache/commitizen-npm-4.2.4-6b34777726-5b0ae7310e.zip/node_modules/commitizen/bin/git-cz.js
- /Users/lgilbraith/code/dt-fleet-micro-fe-albus/.yarn/cache/commitizen-npm-4.2.4-6b34777726-5b0ae7310e.zip/node_modules/commitizen/bin/git-cz
Aborting commit due to empty commit message.
```

**Comment by juliuscc** [comment]
Why would it be any different if we use npm vs yarn? Either the package should be there or it is not? I am a bit confused of how the new version of yarn works 😅

**Comment by laytong** [comment]
Essentially it's a new way of doing module resolution/installation, instead of the current way npm does things.
https://next.yarnpkg.com/features/pnp

As a part of this it does a lot of dependency resolution and for whatever reason this inquirer package is not declared correctly by some package, but it's not clear to me if it's this package or a dependency of this package

**Comment by juliuscc** [comment]
TBH I think we should add Inquirer as a dependency, because [we do use it](https://github.com/digitalroute/cz-conventional-changelog-for-jira/blob/master/LimitedInputPrompt.js#L2). Could you create a PR and test if that solves your problem?

**Comment by laytong** [comment]
It's interesting that it's in the package lock/available for import when it's not specified in the package.json?

I'll hopefully have some time to test this soon, but it feels like a pretty safe addition since you're actually using it?

**Comment by Saeger** [comment]
Fixed by #53
